### ISSUE TITLE
Idea of Molecular Dynamics

### ISSUE THREAD

**Issue by BetaFoprhoton** [petrolpark/Destroy#434]
Firstly, I'd like to thank destroy that made me fell in love with the wonderful chemistry world. And I really like this mod. :)

# Description
Destroy seems to be using **the vanilla crafting recipe (input-(cat.)-output)** for calculating chemical reactions. This has many **benefits**, such as fast speed, clear reactants and products in chemical reactions, and so on.
However, this method also has its **shortcomings**, such as:
1. Due to the fact that the chemical reaction of Destroy is calculated by **pre-set reaction recipes**, it cannot include all types of reactions.
2. The vanilla input-output recipes cannot well reflect the microscopic quantum world. The behavior of particles, such as luminol and iodine clock reaction **(maybe)**. 

So I wonder if **computational chemistry** or further **(molecular dynamics)** can be introduced into destroy:
**Pros:**
1. It can include **most of the chemical reactions**, and the **reacting rate, temperature, molecular diffusion**, etc. are more reasonable.
2. There are many libs available in the community (such as **deepmd-kit, mdanalysis**, and some libs written in Java and Kotlin), but it is unclear whether these libs are suitable for Destroy.
3. Able to demonstrate the mechanism and process of chemical reactions. (explained in detail later)
...

**Cons:**
1. If the community libs (**espeically** those written in python, need **embeddable python venv**) is used to include it in the project, may be a situation where the mod file will be too big.
2. There is a certain demand for the computing power of players' computers.
3. Simulating chemical reactions through calculations may lead to overly complex reaction systems.
4. It is difficult to show these simulated chemical reactions on jei. (vanilla recipes)
...

### Global simulation (requiring significant computing power)
1. Each vat has a unique simulation of molecular reactions universe, processed in parallel.
2. It is not necessary to fully simulate, and the number of material molecules inside the vat can be proportionally reduced to reduce the computational power required
3. Scan the universe, record molecular, temperature, etc. of the universe, and then give these results back to the vat
### Partial simulation
1. When **the reaction system conditions inside the vat change**, create a simulated universe and initialize it as a sample of the system inside the vat.
2. Regularly correct the vat based on the simulation results, and maybe deep learning can be used to make the calculation of vat more real (for some vat that runs fixed and simple tasks, it might can make its calculation speed faster)
﻿
### Ponder that explains the mechanism of chemical reactions:
The Create mod's Ponder is simple, easy to understand, allowing players to quickly understand the gameplay of mods. I wonder if it is possible to introduce a ponder that specifically introduces the mechanism of chemical reactions, using entity molecules and text to explain the reaction mechanisms of the generic reactions. This may not only make player interested in chemistry but also be used for some educational purposes.

### Other
I'm not very familiar with molecular dynamics, whether it is feasible depends on destroy's situation.  :)

## Minecraft Username
BetaFoprhoton ([already suggested before](https://github.com/petrolpark/Destroy/issues/352))

**Comment by petrolpark** [comment]
This is already what Destroy does.

Ponder is not sufficient for explaining chemical reactions and an upcoming update focuses on education.

**Comment by TheRedCrafter344** [comment]
what destroy does is closer to pre-set recipes than this suggestion, so the above answer is a bit misleading.

**Comment by petrolpark** [comment]
Quantum simulation is not in scope.
